resources/common/plugins/template/cvvc-oto-gen/cvvc-oto-gen.js

```diff
@@ -117,136 +112,322 @@ let aliasCountMap = new Map()
+    push(entry, "CV")
+}
+
+function pushVV(sample, index, alias, nextHasConsonant) {
+    let thisAlias = checkAliasCount(alias, false, false)
+
+    let start = offset + index * beatLength - 2*ovlVC
+    let ovl = start + ovlVC
+    let preu = ovl + ovlVC
+    let fixed = preu + fixBuffer
+    let cutoff = 0 - (2*ovlVC + beatLength - ovlVC - (nextHasConsonant ? consLength : 0))
+    
+    let end = start - cutoff
+    let points = [fixed, preu, ovl]
+    points.push(start)
+    let extras = [cutoff.toString()]
+    let notes = new Notes()
+    if (appendTags) {
+        notes.tag = "VV"
+    }
+    let entry = new Entry(sample, thisAlias, start, end, points, extras, notes)
+    push(entry, "CV")
+}
+
+function pushVCV(sample, index, alias, nextHasConsonant) {
+    let thisAlias = checkAliasCount(alias, false, false)
+
+    let start = offset + index * beatLength - consLength - 2*ovlVC
+    let ovl = start + ovlVC
+    let preu = ovl + ovlVC + consLength
+    let fixed = preu + fixBuffer
+    let cutoff = 0 - (2*ovlVC + consLength + beatLength - ovlVC - (nextHasConsonant ? consLength : 0))
+    
+    let end = start - cutoff
+    let points = [fixed, preu, ovl]
+    points.push(start)
+    let extras = [cutoff.toString()]
+    let notes = new Notes()
+    if (appendTags) {
+        notes.tag = "vcv"
+    }
+    let entry = new Entry(sample, thisAlias, start, end, points, extras, notes)
+    push(entry, "other")
+}
+
+function pushTail(sample, index, alias) {
+    let thisAlias = checkAliasCount(alias, false, false)
+
+    let start = offset + index * beatLength - 2*ovlVC
+    let ovl = start + ovlVC
+    let preu = start + 2*ovlVC
+    let fixed = start + 2*ovlVC + 10
+    let cutoff = 0 - (2*ovlVC + 20)
+    
+    let end = start - cutoff
+    let points = [fixed, preu, ovl]
+    points.push(start)
+    let extras = [cutoff.toString()]
+    let notes = new Notes()
+    if (appendTags) {
+        notes.tag = "Tail"
+    }
+    let entry = new Entry(sample, thisAlias, start, end, points, extras, notes)
+    push(entry, "VC")
+}
+
+function pushOther(sample) {
+    let alias = checkAliasCount(sample, true, false)
+
+    let start = offset
+    let ovl = start + 20
+    let preu = start + 10
+    let fixed = start + 30
+    let cutoff = 0 - (40)
+    
+    let end = start - cutoff
+    let points = [fixed, preu, ovl]
+    points.push(start)
+    let extras = [cutoff.toString()]
+    let notes = new Notes()
+    if (appendTags) {
+        notes.tag = "Others"
+    }
+    let entry = new Entry(sample, alias, start, end, points, extras, notes)
+    push(entry, "other")
 }
 
 function parseSample(sample) {
     if (prefix !== "" && !sample.startsWith(prefix)) {
-        pushCV(sample, 0, sample, true, "Others")
+        pushOther(sample)
         return
     }
 
-    let rest = (sample + appendSuffix).slice(prefix.length)
+    // Remove ".wav" extension before appending suffix
+    let rest = (sample.slice(0,-4) + appendSuffix).slice(prefix.length)
```

Comment:
We now support not only wav files, so please use [this API](https://github.com/sdercolin/vlabeler/blob/main/docs/file-api.md#related-global-functions).